src/librustdoc/html/static/search.js

```diff
@@ -254,7 +257,7 @@ window.initSearch = function(rawSearchIndex) {
             });
 
             for (i = 0, len = results.length; i < len; ++i) {
-                var result = results[i];
```

Comment:
Agreed!

Reply:
So, actually, it's an issue because when when you create a `var` inside a loop, its scope is still the whole function. So no, we need to keep this change unfortunately.

Reply:
Oh dang, I was wondering if I was misremembering JS's scoping rules. That sounds right. How about using a different variable name to indicate that the lower `for` loop doesn't depend on output from the upper `for` loop?

Reply:
They're actually both at the same level. `result` is just used in a loop later on. I really miss "normal" scopes when doing JS...

Reply:
Ah, by "lower" and "upper" I meant lower in the file and higher in the file, not in different scopes. But this is not too important.

Reply:
I'll keep things as is then.